### DISCUSSION TITLE
Toggle All/None in Restore panel

### DISCUSSION THREAD

**Discussion by BoneheadFraggle** [home-assistant/frontend#7502] [Supervisor]
It would be very nice with an option to select/deselect all in the restore panel.

When restoring only one add-on for example, it would be neat to only click twice instead of today's 15 times to deselect everyting else.

![image](https://user-images.githubusercontent.com/2287810/97781265-6773a100-1b8a-11eb-9dbc-079844d4c3f8.png)

**Comment by ludeeus** [comment]
How about something in between, like a tree selector?

- [x] Home Assistant
  - [x] Home Assistant 2020.xx.x
- [ ] Folders
  - [ ] Home Assistant configuration
  - [ ] ...
  - [ ] Share
- [ ] Add-ons
  - [ ] AdGuard Home
  - [ ] ...
  - [x] MariaDB
  - [ ] ...

It will not be 2 clicks but will be a lot less than the current situation

**Reply by BoneheadFraggle** [reply]
Personally I don't really see the case where I want to restore all Add-ons, or all folders. So far it's been restore of a single thing or a complete restore. But sure, why not? I'm fine with that. Would still be a great improvement from today 👍

**Reply by matthiasdebaat** [reply]
Just made three different mockups for the snapshot restore dialog. Which one do you prefer?

1. [Closest to the current situation.](https://www.figma.com/proto/CtJmsdwQPxqx951JRYiDfT/Snapshot?page-id=0%3A1&node-id=1%3A199&viewport=-28%2C301%2C0.5634387731552124&scaling=min-zoom) Added option to select/deselect folders and add-ons. Only one button to restore what is selected. Download and Delete snapshot is moved to the vertical dot menu, same as in the lovelace and integration edit. And added icons and images to make the type clearer.

2. [Added the same radio buttons when creating a snapshot, Full and Partial snapshot.](https://www.figma.com/proto/CtJmsdwQPxqx951JRYiDfT/Snapshot?page-id=3%3A291&node-id=3%3A292&viewport=-918%2C388%2C0.7721048593521118&scaling=min-zoom
) When selecting partial the user gets all options deselected. When selecting for example ESPHome the select box of Add-ons will get the intermediate state.

3. [Inspired by Google Backup and Sync software i changed option 2 to a column view. ](https://www.figma.com/proto/CtJmsdwQPxqx951JRYiDfT/Snapshot?page-id=3%3A946&node-id=3%3A947&viewport=-2408%2C252%2C0.8646548390388489&scaling=min-zoom)

**Reply by Misiu** [reply]
@matthiasdebaat personally I like the second option. What will the user see when he had created a partial snapshot and want to restore it? Similar view?

**Reply by matthiasdebaat** [reply]
Good question! In that case you probably want to see what can be restored and skip the 'Partial restore' option

**Reply by BoneheadFraggle** [reply]
I prefer (2). I really like that everything is deselected after choosing Partial. And that it's grouped.

I think you visualized my idea perfectly!

**Reply by matthiasdebaat** [reply]
@Misiu just updated the prototypes with an extra partial snapshot with only one add-on. In this case grouping isn't necessary.

**Comment by BoneheadFraggle** [comment]
Feedback: Just restored an add-on and the new layout and selection structure is just great. Thank you very much!
